core/src/main/java/org/apache/cxf/io/CacheAndWriteOutputStream.java

```diff
@@ -29,7 +29,7 @@
  *
  */
 public class CacheAndWriteOutputStream extends CachedOutputStream {
-
+    private boolean isClosed = false;
```

Comment:
The stream should not be shared between different threads, so we may not need `volatile` or atomic